Extension/src/LanguageServer/client.ts

```diff
@@ -948,13 +1028,22 @@ export class DefaultClient implements Client {
 
                     if (isFirstClient) {
                         workspaceReferences = new refs.ReferencesManager(this);
-
                         // The configurations will not be sent to the language server until the default include paths and frameworks have been set.
                         // The event handlers must be set before this happens.
-                        const inputCompilerDefaults: configs.CompilerDefaults = await languageClient.sendRequest(QueryCompilerDefaultsRequest, {});
+                        const compilerPath: PersistentState<string[]> = new PersistentState<string[]>("CPP.trustedCompilerPaths", []);
+                        const inputCompilerDefaults: configs.CompilerDefaults = await this.requestCompiler(compilerPath.Value);
                         compilerDefaults = inputCompilerDefaults;
                         this.configuration.CompilerDefaults = compilerDefaults;
-
+                        this.configuration.setupConfigurations();
+                        clientCount++;
```

Comment:
The user may add a client later. For example, if the user adds 3 folders, one right after the other, this code will be hit 3 additional times, despite no changes having been made that would result in a trusted compiler being set.  You probably don't want to prompt the user 4 times.  I think you need an `I already did this check` boolean here, to prevent the prompt from appearing multiple times in that scenario.

Reply:
Added a boolean for this in "defaultCompilerFound"